apis/python/src/tiledbsoma/factory.py

```diff
@@ -73,22 +74,22 @@ def _construct_member(
     # Now invoke the appropriate per-class constructor.
     if class_name == "Experiment":
         assert object_type is None or object_type == "group"
-        return Experiment(uri=member_uri, parent=parent, ctx=ctx)
+        return Experiment(uri=member_uri, parent=parent, session_context=session_context)
```

Comment:
@atolopko-czi I made some minor revisions to this big switch in #671 although those changes ended up not being necessary in the ~final approach of that PR, and maybe belong here instead.

The [TileDBObject constructor](https://github.com/single-cell-data/TileDB-SOMA/blob/3ad6cf7cc6e3939851bb77c8ab30c6bf8dcef56d/apis/python/src/tiledbsoma/tiledb_object.py#L28-L32) says to pass in *at most one of* parent or context, because if you supply the parent then the new object inherits the parent context. As currently coded on main, if you pass in both (as we're doing here), the constructor *ignores* the explicitly-provided context in favor of inheriting the parent context. This is pretty confusing so unless there's a case where we need the new object to have a different context than its parent, I suggest adding an assert in the TileDBObject constructor that parent and context are mutually exclusive, and not passing in the context (separately from the parent) here.

Reply:
Just clarifying "unless there's a case" -- I don't know of such a case, but I'm pretty new at this too, so it merits investigation. I do know in #671 that all the CI passes with such an assert.

Reply:
@mlin there's nothing profound here -- I can say this as I wrote the original version :) -- the idea was at each level the normal behavior is take context from parent. With option to have a 'special' per-object context if desired.

Also note what _is_ the parent object -- nominally either a `soma.Experiment`, or a `soma.Collection` consisting of `soma.Experiment` objects. So for `soma.Experiment` in particular it was important to get the context from the parent -- if there is one -- or for it to get its own context since it is the top-level object.

How we handle the case "what if parent and context are both specified" -- yes currently it takes a non-null context as override of the parent. And I like your idea of asserting one XOR the other. (Which is perhaps the way it should have been done all along.)

Reply:
Thanks, both! I'll pass just `parent`, to align with #671. And I'll add the param XOR validation as well.

Reply:
Terrific, thanks @atolopko-czi. I've backed those changes out of #671 which also helps clarify/simplify what's going on there (since they ended up being extraneous).